subsys/bluetooth/mesh/shell/shell_onoff_cli.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+#include <bluetooth/mesh/models.h>
+#include <shell/shell.h>
+
+#include "mesh/net.h"
+#include "mesh/access.h"
+#include "shell_utils.h"
+
+static struct bt_mesh_model *mod;
+
+static int shell_onoff_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (!mod) {
+		if (!shell_model_first_get(BT_MESH_MODEL_ID_GEN_ONOFF_CLI, &mod)) {
+			return -ENODEV;
+		}
+	}
+
+	struct bt_mesh_onoff_cli *cli = mod->user_data;
+	struct bt_mesh_onoff_status rsp;
+
+	int err = bt_mesh_onoff_cli_get(cli, NULL, &rsp);
+
+	if (!err) {
+		shell_print(shell, "Present val: %d, target val: %d, rem time: %d",
+			    rsp.present_on_off, rsp.target_on_off, rsp.remaining_time);
+	}
+
+	return err;
+}
+
+static int onoff_set(const struct shell *shell, size_t argc, char *argv[], bool acked)
+{
+	bool on_off = (bool)strtol(argv[1], NULL, 0);
+	uint32_t time = (argc == 3) ? (uint32_t)strtol(argv[2], NULL, 0) : 0;
+	uint32_t delay = (argc == 4) ? (uint32_t)strtol(argv[3], NULL, 0) : 0;
+
+	if (!mod) {
+		if (!shell_model_first_get(BT_MESH_MODEL_ID_GEN_ONOFF_CLI, &mod)) {
+			return -ENODEV;
+		}
+	}
+
+	struct bt_mesh_onoff_cli *cli = mod->user_data;
+	struct bt_mesh_model_transition trans = { .time = time, .delay = delay };
+	struct bt_mesh_onoff_set set = { .on_off = on_off,
+					 .transition = (argc > 2) ? &trans : NULL };
+
+	if (acked) {
+		struct bt_mesh_onoff_status rsp;
+		int err = bt_mesh_onoff_cli_set(cli, NULL, &set, &rsp);
+
+		if (!err) {
+			shell_print(shell, "Present val: %d, target val: %d, rem time: %d",
+				    rsp.present_on_off, rsp.target_on_off, rsp.remaining_time);
+		}
+
+		return err;
+	} else {
+		return bt_mesh_onoff_cli_set_unack(cli, NULL, &set);
+	}
+}
+
+static int shell_onoff_set(const struct shell *shell, size_t argc, char *argv[])
+{
+	return onoff_set(shell, argc, argv, true);
+}
+
+static int shell_onoff_set_unack(const struct shell *shell, size_t argc, char *argv[])
+{
+	return onoff_set(shell, argc, argv, false);
+}
+
+static int shell_instance_set(const struct shell *shell, size_t argc, char *argv[])
+{
+	struct bt_mesh_model *mod_temp;
+	uint8_t elem_idx = (uint8_t)strtol(argv[1], NULL, 0);
+
+	if (argc == 1) {
+		shell_model_instances_get(shell, BT_MESH_MODEL_ID_GEN_ONOFF_CLI);
+	} else {
+		mod_temp = shell_model_get(BT_MESH_MODEL_ID_GEN_ONOFF_CLI, elem_idx);
+
+		if (mod_temp) {
+			mod = mod_temp;
+		} else {
+			shell_error(shell, "Unable to find model instance for element index %d",
+				    elem_idx);
+		}
+	}
+
+	return 0;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(
+	onoff_cmds, SHELL_CMD_ARG(get, NULL, NULL, shell_onoff_get, 1, 0),
+	SHELL_CMD_ARG(set, NULL, "<onoff> [time] [delay]", shell_onoff_set, 2, 2),
+	SHELL_CMD_ARG(set-unack, NULL, "<onoff> [time] [delay]", shell_onoff_set_unack, 2, 2),
+	SHELL_CMD_ARG(instance-set, NULL,
```

Comment:
I would not recommend include 2 commands in one. A user calls `mesh-onoff instance-set` without arguments, but it returns list of models instances instead of what are the parameters for this command.